jax/numpy/linalg.py

```diff
@@ -61,6 +61,19 @@ def svd(a, full_matrices=True, compute_uv=True):
   return lax_linalg.svd(a, full_matrices, compute_uv)
 
 
+@_wraps(onp.linalg.matrix_rank)
+def matrix_rank(M, tol=None):
+  M = _promote_arg_dtypes(np.asarray(M))
+  if M.ndim > 2:
+    raise TypeError("array should have 2 or fewer dimensions")
+  if M.ndim < 2:
+    return np.any(M != 0).astype(np.int32)
```

Comment:
Casting to `np.int32` can cause issues if we are testing for `dtype` in `tests/linalg_test::testMatrixRank`. To get around this we are not checking for `dtype` in that test.